### PR TITLE
Radiance sensor from view

### PR DESCRIPTION
**Pull request by gmoysan**
New application to create a radiance sensor (type=observer) based on the 3D view:  GetActiveWindow().GetCameraFrame() is used to get the field of view of the window, and use these values to set the field of view of the radiance sensor
Test added in a separate Push (run_test_03_speos_sensors.py)

### CODE REVIEW

**Comment by StefanThoene** [comment]
@gmoysan All unit test are failing there seems to be an issue with the change you have done...

**Comment by pluAtAnsys** [comment]
> @gmoysan All unit test are failing there seems to be an issue with the change you have done...

I can see the problems coming from the modifications on base.py. I am not think they are needed, no?

**Comment by gmoysan** [comment]
> 

@pluAtAnsys correct, the base.py changes are not required

**Comment by pluAtAnsys** [comment]
> > 
> 
> @pluAtAnsys correct, the base.py changes are not required

I think that should solve the unit test fail problem

**Comment by gmoysan** [comment]
> > > 
> > 
> > 
> > @pluAtAnsys correct, the base.py changes are not required
> 
> I think that should solve the unit test fail problem

@pluAtAnsys how do I exclude the changes to base.py ? Do I need to create a new push or can I modify this one ?

**Comment by pluAtAnsys** [comment]
> > > > 
> > > 
> > > 
> > > @pluAtAnsys correct, the base.py changes are not required
> > 
> > 
> > I think that should solve the unit test fail problem
> 
> @pluAtAnsys how do I exclude the changes to base.py ? Do I need to create a new push or can I modify this one ?

yes, you will need to create a new push to over-write your previous one.

File: ansys_optical_automation/application/radiance_from_view.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    InputHelper.PauseAndGetInput("Adjust the window to the desired point of view and validate")
+    Camera = GetActiveWindow().GetCameraFrame()
+    HFOV = round(GetActiveWindow().HorizontalFieldOfView * 180 / 3.14, 0)
+    VFOV = round(GetActiveWindow().VerticalFieldOfView * 180 / 3.14, 0)
+    DatumCS = DatumOriginCreator.Create(Camera)
+    cs = DatumCS.CreatedOrigin
+    cs.SetName("ObserverOriginRadianceSensor")
+
+    Radiance = RadianceSensor("Radiance_from_View", SpeosSim, SpaceClaim)
+    Radiance.set_observer_type("observer")
```
Comment (pluAtAnsys) [inline ansys_optical_automation/application/radiance_from_view.py:15]: I believe you would like to use radiance sensor DefinitionFrom method not observer_type method, as Frame with observer type radiance sensor does not have front & upper direction.

File: ansys_optical_automation/application/radiance_from_view.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from ansys_optical_automation.speos_process.speos_sensors import RadianceSensor
+
+
+def main():
+    Command.Execute("FlyThroughCamera")
```
Comment (pluAtAnsys) [inline ansys_optical_automation/application/radiance_from_view.py:5]: Error "Command" is not defined when running from Ansys Speos, you will need Command back to the base.py

File: ansys_optical_automation/application/radiance_from_view.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    Camera = GetActiveWindow().GetCameraFrame()
+    HFOV = round(GetActiveWindow().HorizontalFieldOfView * 180 / 3.14, 0)
+    VFOV = round(GetActiveWindow().VerticalFieldOfView * 180 / 3.14, 0)
+    DatumCS = DatumOriginCreator.Create(Camera)
+    cs = DatumCS.CreatedOrigin
+    cs.SetName("ObserverOriginRadianceSensor")
+
+    Radiance = RadianceSensor("Radiance_from_View", SpeosSim, SpaceClaim)
+    Radiance.set_observer_type("observer")
+    Radiance.set_type("colorimetric")
```
Comment (pluAtAnsys) [inline ansys_optical_automation/application/radiance_from_view.py:16]: in-correct result running the script, "colorimetric" has not been used to define sensor

**Review by pluAtAnsys — CHANGES_REQUESTED** [review]

File: ansys_optical_automation/application/radiance_from_view.py (hunk 2 above)
Comment (StefanThoene) [reply ansys_optical_automation/application/radiance_from_view.py]: i added a fix for it

File: ansys_optical_automation/application/radiance_from_view.py (hunk 1 above)
Comment (StefanThoene) [reply ansys_optical_automation/application/radiance_from_view.py]: i fixed this by adding the additional method on sensor class
unit test is missing for this one

File: ansys_optical_automation/application/radiance_from_view.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import math
+
+# import sys
+# repo_path = r"your_repo_path"
+# sys.path.append(repo_path)
+from ansys_optical_automation.speos_process.speos_sensors import RadianceSensor
+
+
+def main():
+    SpaceClaim.Api.V21.Command.Execute("FlyThroughCamera")
```
Comment (pluAtAnsys) [inline ansys_optical_automation/application/radiance_from_view.py:11]: I do not think using a specific version here is a good idea.

**Review by pluAtAnsys — CHANGES_REQUESTED** [review]

File: ansys_optical_automation/application/radiance_from_view.py (hunk 4 above)
Comment (pluAtAnsys) [reply ansys_optical_automation/application/radiance_from_view.py:11]: Solution can be define self.Command = scdm_api.Command in the base.py and 
```suggestion
    Radiance = RadianceSensor("Radiance_from_View", SpeosSim, SpaceClaim)
    Radiance.Command.Execute("FlyThroughCamera")
```

Comment (StefanThoene) [reply ansys_optical_automation/application/radiance_from_view.py:11]: @gmoysan do you want to give it a try or shall i do it?

Comment (gmoysan) [reply ansys_optical_automation/application/radiance_from_view.py:11]: @StefanThoene how can I get access to your changes (lines 1-6 for example) to test ? I believe your changes have not been pushed to the main, so I can't clone the main to access them.

Comment (pluAtAnsys) [reply ansys_optical_automation/application/radiance_from_view.py:11]: @gmoysan you can "pull" or type "git pull" at pycharm terminal to get the changes from Stefan

File: ansys_optical_automation/application/radiance_from_view.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    Radiance.set_fov(HFOV, VFOV, HFOV * 10, VFOV * 10)
+
+    MessageBox.Show(
+        "A Radiance Sensor was created with "
+        + str(int(HFOV))
+        + "°x"
+        + str(int(VFOV))
+        + r"° field of view \n and 0.1° central resolution"
+    )
+    SpaceClaim.Api.V21.Command.Execute("FlyThroughCamera")
```
Comment (pluAtAnsys) [inline ansys_optical_automation/application/radiance_from_view.py:34]:
```suggestion
   Radiance.Command.Execute("FlyThroughCamera")
```

File: ansys_optical_automation/application/radiance_from_view.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    SpaceClaim.Api.V21.Command.Execute("OrthographicCamera")
```
Comment (pluAtAnsys) [inline ansys_optical_automation/application/radiance_from_view.py:35]:
```suggestion
    Radiance.Command.Execute("OrthographicCamera")
```

File: ansys_optical_automation/speos_process/speos_sensors.py [hunk 7]
```diff
@@ -498,6 +498,16 @@ def set_observer_type(self, observer_type):
             error_message = "please provide a radiance type as observer or focal"
             raise ValueError(error_message)
 
+    def set_definition_type(self, definition_type):
```
Comment (pluAtAnsys) [inline ansys_optical_automation/speos_process/speos_sensors.py:501]: please add the missing docstring

**Review by pluAtAnsys — CHANGES_REQUESTED** [review]

**Review by pluAtAnsys — APPROVED** [review]

**Pull request merged**